### DISCUSSION TITLE
Wires take too long to render.

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#12904] [Ideas and suggestions]
Hello. After doing some testing on my sub as to why all the components on my ship where slowing things down so much (160fps with dugong, 60fps at best with my ship, much lower on systems with lesser hardware), I found that selecting all wires and selecting 'hide in game' boosts FPS to nearly 100. Drawmap time in particular went from 6~7ms down to 4ms.

I suggest looking into your logic for rendering the wires. IMO it should be a giant vertex buffer containing all wires on a single ship in one draw call, using vertex colors for wire color. Culling outside of 'per ship' shouldn't be needed (or used) since making multiple draw calls is likely gonna hurt more then drawing a few thousand vertices off screen. And said vertex buffer should only be reconstructed when a wiring change is made.

**Comment by deltanedas** [comment]
In addition, only modify the big buffer when rewiring is complete, have a seperate one that is cheap to modify while rewiring.

**Comment by BlackMoons** [comment]
Swinging disconnected wires ends should also be a separate buffer/rendering since they are a minority.

They could be done in a custom vertex shader but it seems overkill when most ships will have a dozen disconnected wires at most.

**Comment by deltanedas** [comment]
was this implemented in unstable?

**Comment by Regalis11** [comment]
We decided to close the ticket because it's not really a bug report, but more of a suggestion. The suggestion is good though, I think we might be looking into it at some point.

**Comment by deltanedas** [comment]
move to suggestions then so it can be tracked easier :trollface:
